addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -647,74 +659,63 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                        @Override
-                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                            listener.yearChangeEvent(mac, decode(value));
-                        }
-                    });
+                    year = messagePart.substring("year%3A".length());
                 }
-                // Parameter Artwork
-                else if (messagePart.startsWith("artwork_track_id%3A")) {
-                    String url = messagePart.substring("artwork_track_id%3A".length());
-                    // NOTE: what is returned if not an artwork id? i.e. if a
-                    // space?
-                    if (!url.startsWith(" ")) {
-                        url = "http://" + host + ":" + webport + "/music/" + url + "/cover.jpg";
+                // Parameter Artwork_url
+                else if (messagePart.startsWith("artwork_url%3A")
+                        && !messagePart.startsWith("artwork_url%3Ahttp%3A%2F%2F")) {
+                    url = messagePart.substring("artwork_url%3A".length());
+                    if (url.startsWith("%2F")) {
+                        url = "http://" + host + ":" + webport + decode(url);
+                    } else {
+                        url = "http://" + host + ":" + webport + "/" + decode(url);
                     }
-                    final String value = url;
-                    updatePlayer(new PlayerUpdateEvent() {
-                        @Override
-                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                            listener.coverArtChangeEvent(mac, decode(value));
-                        }
-                    });
                 } else {
                     // Added to be able to see additional status message types
                     logger.trace("Unhandled status message type '{}'", messagePart);
                 }
             }
+            // update cover art
+            // update remote title
+            // update artist
+            // update album
+            // update genre
+            // update year
+
+            final String f_url = url;
+            final String f_remoteTitle = remoteTitle;
+            final String f_artist = artist;
+            final String f_album = album;
+            final String f_genre = genre;
+            final String f_year = year;
+
+            updatePlayer(new PlayerUpdateEvent() {
+                @Override
+                public void updateListener(SqueezeBoxPlayerEventListener listener) {
+                    listener.coverArtChangeEvent(mac, f_url);
+                    listener.remoteTitleChangeEvent(mac, decode(f_remoteTitle));
+                    listener.artistChangeEvent(mac, decode(f_artist));
+                    listener.albumChangeEvent(mac, decode(f_album));
+                    listener.genreChangeEvent(mac, decode(f_genre));
+                    listener.yearChangeEvent(mac, decode(f_year));
```

Comment:
You bundle almost all the song metadata together (remoteTitle, artist, album, genre, and year), but you excluded title from this bundling.  Is there a specific reason for excluding title?

Reply:
Yes because title are always in the message. No matter what kind of music (web radio, local music etc..) is played by the squeezebox server. It's not the case of artist, album, genre, remote title, year. When a new message comes up, these variable are initialised with a empty string. If a variable is present in the message it will be updated, else it remains empty